qndiag/qndiag.py

```diff
@@ -128,22 +134,29 @@ def qndiag(C, B0=None, weights=None, max_iter=1000, tol=1e-6,
         # Gradient
         G = np.average(D / diagonals[:, :, None], weights=weights_,
                        axis=0) - np.eye(n_features)
+        if ortho:
+            G = G - G.T
         g_norm = np.linalg.norm(G)
         if g_norm < tol * np.sqrt(n_features):  # rescale by identity
             break
 
         # Hessian coefficients
         h = np.average(diagonals[:, None, :] / diagonals[:, :, None],
-                       weights=weights_, axis=0)
-        # Quasi-Newton's direction
-        det = h * h.T - 1.
-        det[det < lambda_min] = lambda_min  # Regularize
-        direction = -(G * h.T - G.T) / det
+                    weights=weights_, axis=0)
+        if ortho:
+            det = h + h.T - 1
+            det[det < lambda_min] = lambda_min  # Regularize
+            direction = -G / det
+        else:
+            # Quasi-Newton's direction
+            det = h * h.T - 1.
+            det[det < lambda_min] = lambda_min  # Regularize
+            direction = -(G * h.T - G.T) / det
```

Comment:
```suggestion
        det = h + h.T - 1
        det[det < lambda_min] = lambda_min  # Regularize
        if ortho:
            direction = -G / det
        else:
            direction = -(G * h.T - G.T) / det
```

Reply:
Wait it is not the same thing there. you h + h.T - 1 or h * h.T - 1